opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -725,8 +739,28 @@ private void addFile(File file, String path, Ctags ctags)
         fa.setFoldingEnabled(env.isFoldingEnabled());
 
         Document doc = new Document();
-        try (Writer xrefOut = newXrefWriter(fa, path)) {
+        CountingWriter xrefOut = null;
+        try {
+            String xrefAbs = null;
+            File transientXref = null;
+            if (env.isGenerateHtml()) {
+                xrefAbs = getXrefPath(path);
+                transientXref = new File(TandemPath.join(xrefAbs,
+                        PendingFileCompleter.PENDING_EXTENSION));
+                xrefOut = newXrefWriter(path, transientXref, env.isCompressXref());
+            }
+
             analyzerGuru.populateDocument(doc, file, path, fa, xrefOut);
+
+            // Avoid producing empty xref files.
+            if (xrefOut != null && xrefOut.getCount() > 0) {
+                PendingFileRenaming ren = new PendingFileRenaming(xrefAbs,
+                        transientXref.getAbsolutePath());
+                completer.add(ren);
+            } else if (xrefOut != null) {
+                LOGGER.log(Level.FINER, "xref for {0} would be empty, removing", path);
+                transientXref.delete();
```

Comment:
I worry on NTFS that a deletion of an open file wouldn't succeed as it would do on *nix. Perhaps instead schedule a `PendingFileDeletion` of the `transientXref`?

Reply:
Good point. I had the same idea however thought it is unnecessary.